tensilelite/Tensile/Common.py

```diff
@@ -1609,6 +1585,39 @@ def assignGlobalParameters( config ):
 
   global globalParameters
 
+  if config["ValidateMatrixInstructions"]:
+
+    globalParameters["ValidateMatrixInstructions"] = config["ValidateMatrixInstructions"]
+
+    print1("Matrix instruction validation enabled")
+
+    validMFMA["_format9"] = []
+
+    for MFMA in [validMFMA["H"], validMFMA["S"], validMFMA["B"], validMFMA["D"], validMFMA["X"], validMFMA["F8"], validWMMA]:
+      for MI in MFMA:
+        for bm in range(int(math.log(MI[3],2))+1):
+          for tt0 in range(1,validTT+1):
+            for tt1 in range(1,validTT+1):
+              for wave_m in range (3):
+                for wave_n in range(3):
+                  validMFMA["_format9"].append([MI[0],MI[1],MI[2],MI[3],2**bm,tt0,tt1,2**wave_m, 2**wave_n])
+    validMatrixInstructions = [[], [-1]] + validMFMA["H"] + validMFMA["S"] + validMFMA["B"] + validMFMA["D"] + validMFMA["B1k"] + validMFMA["X"]
+    validMatrixInstructions = validMatrixInstructions + validMFMA["_format9"]
+    
+    validSMFMA["_format9"] = []
+    for SMFMA in [validSMFMA["H"], validSMFMA["B"], validSMFMA["4xi8"], validSMFMA["F8"]]:
+      for MI in SMFMA:
+        for bm in range(int(math.log(MI[3],2))+1):
+          for tt0 in range(1,validTT+1):
+            for tt1 in range(1,validTT+1):
+              for wave_m in range (3):
+                for wave_n in range(3):
+                  validSMFMA["_format9"].append([MI[0],MI[1],MI[2],MI[3],2**bm,tt0,tt1,2**wave_m, 2**wave_n])
+    validSparseMatrixInstructions = validSMFMA["H"] + validSMFMA["B"] + validSMFMA["4xi8"]
+    validMatrixInstructions = validMatrixInstructions + validSparseMatrixInstructions + validSMFMA["_format9"]
+
+    validParameters["MatrixInstruction"] = validMatrixInstructions
```

Comment:
Create a function `populateValidMatrixInstructions` for better encapsulation.

Reply:
Good catch!